### DISCUSSION TITLE
Using `cfnText` on Tags to handle Fn

### DISCUSSION THREAD

**Discussion by WolfangAukang** [jcouyang/dhall-aws-cloudformation#12] [General]
I have been trying this library and understanding how to replicate some Cloudformation templates into it. I found the following obstacle:

There are times I use `!Sub` on tags like this:
```
-   Key: Name
    Value: !Sub "${AWS::StackName}.mydomain.com"
```

The problem I see is that both key and value are defined as Text, [according to `Tag.dhall`](https://github.com/jcouyang/dhall-aws-cloudformation/blob/27b658e0f226c7a3bac1c43f9e9b7b2bf18701cc/cloudformation/Tag.dhall#L1). I created a fork where I [replaced the type with `CfnText`](https://github.com/WolfangAukang/dhall-aws-cloudformation/commit/4912cd9d56659007bebf06a23e8d1ba53b2a82d9), so it is possible to perform this kind of operations into tags.

Is there any reason to use `Text` instead of `CfnText` for tags?

**Comment by jcouyang** [comment]
not often use Tag myself but from the doc can't tell it could be intrinsic fn https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html

it kind of make sense to me that Value can be intrinsic fn, but does `Key` also support that?

**Reply by WolfangAukang** [reply]
Yes, at least if I try to use `!Sub` on the key, I have no complaints on CloudFormation. I know that logically it wouldn't make a lot of sense to use an operation on the key field, but it is possible.

**Comment by jcouyang** [comment]
btw, `Tag.dhall` is generated from code, so the correct place should be somewhere between https://github.com/jcouyang/dhall-aws-cloudformation/blob/27b658e0f226c7a3bac1c43f9e9b7b2bf18701cc/src/Dhall/Cloudformation.hs#L173-L174

pull request is welcome if you are keen to make the change
otherwise i can update it later when free

**Reply by WolfangAukang** [reply]
Perfect, I can create the PR
